src/telliot_feeds/reporters/tellor_360.py

```diff
@@ -274,20 +266,25 @@ async def fetch_datafeed(self) -> Optional[DataFeed[Any]]:
         available tips for the datafeed unless the user has not selected a query tag or
         used the random feeds flag.
         """
+        # reset autopay tip every time fetch_datafeed is called
+        # so that tip is checked fresh every time and not carry older tips
+        self.autopaytip = 0
+        # TODO: This should be removed and moved to profit check method perhaps
+        if self.check_rewards:
+            # calculate tbr and
+            _ = await self.rewards()
+
         if self.use_random_feeds:
             self.datafeed = suggest_random_feed()
 
         # Fetch datafeed based on whichever is most funded in the AutoPay contract
         if self.datafeed is None:
             suggested_feed, tip_amount = await get_feed_and_tip(self.autopay)
 
-            if suggested_feed is not None:
-                self.autopaytip = tip_amount  # type: ignore
+            if suggested_feed is not None and tip_amount is not None:
+                self.autopaytip += tip_amount
+
                 self.datafeed = suggested_feed
                 return self.datafeed
 
-        # TODO: This should be removed and moved to profit check method perhaps
```

Comment:
@oraclown, this is what I wanted to ask you about if you knew why this was put here and see if you can spot breaks from this change.

Reply:
sorry this is for the lines below

Reply:
I think I just put the TODO comment there. I was saying that it's weird that we do profit calc stuff in the fetch datafeed function. I'm not sure what happens if you reset `autopaytip` each time, or if this will cause unintended breaks.